build.sbt

```diff
@@ -713,36 +713,39 @@ lazy val compiler = configureAsSubproject(project)
   .dependsOn(library, reflect)
 
 lazy val interactive = configureAsSubproject(project)
+  .settings(commonSettings)
   .settings(disableDocs)
   .settings(skip in publish := true)
   .settings(
     name := "scala-compiler-interactive",
-    description := "Scala Interactive Compiler"
+    description := "Scala Interactive Compiler",
+    // TODO remove `-unit-special` once we re-STARR on 2.13.3, because scala/bug#11965
+    scalacOptions in Compile ++= Seq("-Xlint:-unit-special,_", "-unchecked", "-Werror", "-Wconf:cat=deprecation:msg=early.initializers:s"),
   )
   .dependsOn(compiler)
 
 lazy val repl = configureAsSubproject(project)
+  .settings(commonSettings)
   .settings(disableDocs)
   .settings(skip in publish := true)
-  .settings(scalacOptions in Compile += "-Xlint:-deprecation,-inaccessible,-nonlocal-return,-valpattern,-doc-detached,_")
+  .settings(scalacOptions in Compile ++= Seq("-Xlint", "-Werror", "-unchecked", "-Wconf:cat=deprecation:msg=early.initializers:s"))
   .dependsOn(compiler, interactive)
 
 lazy val replFrontend = configureAsSubproject(Project("repl-frontend", file(".") / "src" / "repl-frontend"))
+  .settings(commonSettings)
   .settings(disableDocs)
   .settings(skip in publish := true)
   .settings(
     libraryDependencies ++= jlineDeps,
     name := "scala-repl-frontend",
-    scalacOptions in Compile += "-Xlint:-inaccessible,-nonlocal-return,-valpattern,-doc-detached,_",
+    scalacOptions in Compile ++= Seq("-Xlint", "-Werror", "-unchecked"),
```

Comment:
Noice. On the recent linting PR, I noticed that I had pushed these exclusions down to subprojects and they deserved review.

Reply:
yeah, I figured. we'll get to it, a project at a time and a flag at a time